src/db/mysql/batch.rs

```diff
@@ -34,92 +50,105 @@ pub fn create(db: &MysqlDb, params: params::CreateBatch) -> Result<results::Crea
                 _ => e.into(),
             }
         })?;
-    Ok(encode_id(timestamp))
+
+    do_append(db, batch_id, params.user_id, collection_id, params.bsos)?;
+
+    Ok(encode_id(batch_id))
 }
 
 pub fn validate(db: &MysqlDb, params: params::ValidateBatch) -> Result<bool> {
```

Comment:
I was going to ask if it made more sense to do a fetch of all the `batch_uploads.batch_id`s for a given user and then compare them against the incoming list vs. doing a ton of `select 1 where...` calls, but I see that this is a long standing thing.

Reply:
this just validates the batch itself: so I wouldn't say there's a ton of them